### PR TITLE
TC 1 Elevation Final Edge Case & Loading State Implementation

### PR DESCRIPTION
**Pull request by fiyinfoluwaafol**
### Description
This PR essentially consists of finishing up the final edge case for TC 1, which goes as described below:
User 1 syncs their google calendar
User 2 syncs their google calendar
User 1 updates a shared event among them on google calendar
User 1 re-syncs their google calendar
The events for both User 1 and User 2 on  app should by updated

Additionally for the sake of showing that my loading state works, I added a 2 second simulate delay (just for the purpose of the demo). Also added a horizontal scroll for the recommended times such that they are also clickable buttons.

Subsequent PRs will address other base requirements such as interesting cursor interaction (possibly a custom tooltip upon hover of calendar events)

### Milestones
- This contributes to the user story of the being able to get recommended times to schedule events
- As well as visual polish as it relates to giving the user feedback in the form of a loading state that the action is being performed.

### Test Plan
Below are videos showcasing the discussed features in the description:
[Horizontal Scroll, Loading State & Final TC 1 Edge Case](https://www.loom.com/share/8b07779d98c94ae5a21f70f450ad0864?sid=8e79902b-721e-416e-a33c-289d4ae1f43a)
[Addition to TC 1 Edge Case](https://www.loom.com/share/5110cd49bfde469b961d037e1c0fc89e?sid=3897e4b1-ba43-41e3-8194-c5afc5740c88)

### CODE REVIEW

File: backend/recommendation.js [hunk 1]
```diff
@@ -62,24 +67,22 @@ async function recommendEventSlots(userId, duration, targetDate, invitees = [])
...
+                timeGrid[i].available = false;
             }
-            timeGrid = timeGrid.map((slot, index) => ({ slot: index, available: slot.available && userTimeGrid[index].available }));
         });
 
         const availableSlots = findAvailableSlots(timeGrid, duration);
         if (availableSlots.length === 0) {
-            throw new Error('No available slots found');
+            return -1;
         }
```
Comment (kate-luckerman) [inline backend/recommendation.js:86]: Introducing this -1 value just adds confusion, it would be clearer to just handle the case that the length is 0 in the frontend.

File: frontend/src/utils/utils.js [hunk 2]
```diff
@@ -275,7 +276,7 @@ async function fetchRecommendations(setRecommendations, duration, participants,
     body: JSON.stringify({
         duration,
         invitees: participants.split(',').map(email => email.trim()).filter(email => email !== ''),
-        targetDate: getTomorrowsDate(date)
```
Comment (kate-luckerman) [inline frontend/src/utils/utils.js:278]: is the getTomorrowsDate function being used anywhere anymore?
